edgelet/iotedged/src/windows.rs

```diff
@@ -64,9 +68,16 @@ fn run_as_service(_: Vec<OsString>) -> Result<(), Error> {
     let settings = app::init_win_svc()?;
     let main = super::Main::new(settings);
     let shutdown_signal = signal::shutdown()
-        .select(receiver.into_future().map(|_| ()).map_err(|_| ()))
-        .map(|_| ())
-        .map_err(|_| ());
+        .select(receiver.map_err(|_| ()))
+        .map(move |_| {
+            info!("Stopping {} service.", IOTEDGED_SERVICE_NAME);
+            let _ = update_service_state(status_handle, ServiceState::StopPending).map_err(|err| {
```

Comment:
Curious: What happens if the `let _ = ` assignment isn't done and we instead return what `map_err` returns?

Reply:
Do we ever send a `ServiceState::Stopped`?

Reply:
>Curious: What happens if the `let _ = ` assignment isn't done and we instead return what `map_err` returns?

The result of `select()` on line 71 has a complicated error type `(Error, OtherFuture)`  which a `map` / `and_then` on line 72 can't construct.

I can change it to a `match` so that it doesn't need the `let _ = `

>Do we ever send a `ServiceState::Stopped`?

Line 91.